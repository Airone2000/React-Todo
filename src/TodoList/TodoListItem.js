import React from 'react';
import PropTypes from 'prop-types';

function TodoListItem({todo, onDeleteTodo, onToggleTodo}) {
  const {title} = todo;
  return (
    <div className="todolist-item">
      <input 
        type="checkbox"
        onChange={onToggleTodo}
        checked={todo.done}
      />
      <div className={'todo-title ' + (todo.done && 'done')}>
        {title}
      </div>
      <button 
        className="todo-delete-btn"
        onClick={onDeleteTodo}
      >
        -Delete
      </button>
    </div>
  );
}

TodoListItem.propTypes = {
  todo: PropTypes.shape({
    title: PropTypes.string.isRequired
  }).isRequired
};


export default TodoListItem;