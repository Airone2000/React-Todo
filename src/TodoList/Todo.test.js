import React from 'react';
import renderer from 'react-test-renderer';
import TodoListItem from './TodoListItem';

it('renders a TodoListItem', () => {

  const props = {
    todo: {title: 'Une todo', done: true},
    onDeleteTodo: function(){},
    onToggleTodo: function(){}
  };

  const tree = renderer
    .create(<TodoListItem todo={props.todo} onDeleteTodo={props.onDeleteTodo} onToggleTodo={props.onToggleTodo} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});