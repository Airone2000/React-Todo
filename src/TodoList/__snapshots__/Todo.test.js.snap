// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a TodoListItem 1`] = `
<div
  className="todolist-item"
>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="todo-title done"
  >
    Une todo
  </div>
  <button
    className="todo-delete-btn"
    onClick={[Function]}
  >
    -Delete
  </button>
</div>
`;
